### DISCUSSION TITLE
Is it possible to use multiple ManagetMqttClients in 1 application?

### DISCUSSION THREAD

**Discussion by ondrej-fabianek** [dotnet/MQTTnet#1725] [Q&A]
Hello, 
I would like to ask if it is possible to use multiple _ManagedMqttClients_ in a single application (each connecting to a different broker).

I am asking because in my application if I do that, the clients start generating lots of "_MqttConnectingFailedException_" and  "_MqttCommunicationTimedOutException_" errors. If I start just one of the clients, it works ok. Since those 2 clients run in different hosted services and are created as independently as possible, I am very confused why is that happening.

I will be very grateful for any advice. 

(I am using version 4.2.0.706, but the same thing was happening in version 3.x)

**Comment by chkr1011** [comment]
Having multiple clients in one application is supported. We also have integration tests for that  Do you have a code snipped you can share which reproduces the error?

**Reply by SeppPenner** [reply]
The only thing to notice should be that all client identifiers must be unique in this scenario @ondrej-fabianek.

**Comment by ondrej-fabianek** [comment]
Hi, 
thank you for the response. It turned out the problem was in our network infrastructure, not related to the MTTnet.
